### DISCUSSION TITLE
Mascot for Rust UCANs

### DISCUSSION THREAD

**Discussion by cdata** [ucan-wg/spec#71] [General]
Hey folks, maintainer of [cdata/rs-ucan](https://github.com/cdata/rs-ucan) here.

On the community call today there was a mention that we still need a Rust UCAN mascot. For your consideration I would like to introduce you to the official Rust mascot Ferris. Ferris is a happy crab who brings joy to Rustaceans everywhere.

**Importantly:** you can find a wealth of visual media related to Ferris at https://rustacean.net/.

I would love to help in any way I can to get a UCAN-ified Ferris mascot for rs-ucan. What would be the first steps to get that going?

![cuddlyferris](https://user-images.githubusercontent.com/240083/170388367-827f7604-61c8-4820-ac91-669fa33c7f0d.png)

**Comment by bmann** [comment]
@cdata awesome, thanks for kicking this off.

Basically, write up a "brief"

How do you want Ferris to be combined with a toucan beak and/or a stack of macaroons.

Do you want Ferris to be "as is", or do you want basically a stylized mascot crab inspired by Ferris. 

e.g. maybe one claw up with a toucan beak on it? And the other claw is up with a stack of macaroons? Any ideas / thoughts / references here are welcome.

Fission can sponsor Bruno, the illustrator we work with, to get this done. And I'll point him at this thread.

**Comment by bmann** [comment]
For reference, @b5's go implementation has a gopher mascot https://github.com/ucan-wg/go-ucan#readme

![](https://camo.githubusercontent.com/eaf8810a16c815e1ff923987e3e5efb8ad3c50db1952a7d0ad5bf4623eb53423/68747470733a2f2f697066732e72756e66697373696f6e2e636f6d2f697066732f516d5246586a4d6a564e776e596b69386a4777464268337a6359356d377a6f356f41634e6f7953315053677a41592f7563616e2d676f706865722e706e67)

**Comment by cdata** [comment]
Hi folks, I'm sorry for the delay getting the "brief" together.

I thought it might be fun for Ferris to be wearing the same kind of mask beak as the gopher, but since Ferris doesn't have a nose, maybe they could be wearing it like an awkward hat, off to the side. The macaroons can be spilled out over his shell and/or impaled on the spikes. Here is my embarrassing mock-up of this idea:

<img width="258" alt="ferris-ucan" src="https://user-images.githubusercontent.com/240083/181693006-34421f2d-d0a6-4eda-b25c-d5c9a26ec368.png">

I also like @bmann 's suggestion, if that seems more practical.

WDYT?

**Reply by cdata** [reply]
@bmann

![67347099-f5d43200-f4f5-11e9-9cbd-11e13dfc3738](https://user-images.githubusercontent.com/240083/184735434-c2caf3db-4fa1-47c3-aa98-68ccb3f5c7e7.gif)

**Reply by bmann** [reply]
Yep! It's on the stack for Bruno! Your illustration spot in line for bumped ;)

**Comment by brunomonts** [comment]
Hi @cdata! Nice to meet you and sorry for the delay!

Quick explainer: for an illustration like this, I usually go through a process of creating three different sketches based on your briefing.

This helps us see what direction we would like to take on the next step, and most importantly the direction that we don't want to go for. From those three options, you can choose one of them, or combine more than one, or even ask for a completely different version. So please, feel totally comfortable asking me to add/subtract/change any of it. It's all part of the process. Once we've come up with a final sketch I'll finish it up with colors and details and send it here again so you can give a final check.

Ok, so with that said, here are our first sketches:

![rs-UCAN Mascot - Sketches](https://user-images.githubusercontent.com/65693169/186965161-7500f2a5-3f9b-4e0d-bb07-8e2ab2a672fe.png)

Here I've tried to do concepts based on your and Boris' ideas but with a slightly different Ferris design, a more three-dimensional and "realistic" look (still cute tho :)). And I did a third option that uses the character's original design but adding the toucan's beak and the macarons as well. Pretty simple but in a good way, I hope :)

Please let me know what you think!

**Reply by cdata** [reply]
These are all awesome!

The stacked macaroons in the left-most sketch are more in keeping with the other mascots, so perhaps that's the best way to present them.

I also really like the 3D look in the left-most sketch. The crumbs on the ground in the middle sketch are kind of a cute detail (but maybe optional).

I guess that means my favorite one is the left-most sketch.

Any other votes?

**Reply by brunomonts** [reply]
Awesome! I'll get working on finishing the left-most sketch then.

Thanks for your feedback!

**Comment by brunomonts** [comment]
Okay, I am back with our finished illustration :)

I've done it in vector so we can use the SVG files on the web without a problem. Once you give me your ok I'll upload the files and share them with you.

Please, let me know what you think!


![rs-UCAN Mascot](https://user-images.githubusercontent.com/65693169/189219668-0c049904-b22c-4ab9-8ac0-67ff19847f07.png)

**Reply by cdata** [reply]
@brunomonts it's so beautiful :star_struck:

**Reply by brunomonts** [reply]
awesome, I am so glad you liked it!
